src/components/EmailList/index.jsx

```diff
@@ -0,0 +1,74 @@
+import EmailItem from "../EmailItem";
+import { useEffect, useState } from "react";
+import EmailView from "../EmailView";
+
+function EmailList() {
+  const [emailArray, setEmailArray] = useState(false);
+  const [refreshJson, setRefreshJson] = useState(false);
+  const [currentID, setCurrentID] = useState(0);
+  const [emailItemWidth, setEmailItemWidth] = useState("w-full");
+  const [emailViewWidth, setEmailViewWidth] = useState("w-0");
+  const [buttonClass, setButtonClass] = useState("hidden");
+
+  useEffect(() => {
+    fetch("https://email-client-api.dev.io-academy.uk/emails")
+      .then((response) => response.json())
+      .then((data) => {
+        setEmailArray(data);
+        setRefreshJson(false);
+        return emailArray;
+      });
+  }, [refreshJson]);
+
+  function closeEmail() {
+    setCurrentID(0);
+    setEmailItemWidth("w-full");
+    setEmailViewWidth("w-0");
+    setButtonClass("hidden");
+  }
+
+  function setRead(id) {
+    fetch("https://email-client-api.dev.io-academy.uk/emails/" + id, {
+      method: "PUT",
+    });
+    setRefreshJson(true);
+  }
+
+  return (
+    <div>
+      <div className="flex">
+        <div className="md:w-2/6 overflow-scroll max-h-screen">
+          {emailArray && (
+            <div>
```

Comment:
```suggestion
            <div className={`${emailItemWidth} md:w-full`}>
```
You can set the `emailItemWidth` (really the `emailListWidth` if you put it here) classes here on the list's container instead of on each email item to reduce some code complexity and save on the amount of redrawing / total number of DOM changes you cause when you toggle this state.

An elegant solution to the problem of hiding/showing the emails though!